File: app/src/main/java/com/hfad/beeradviser/Java/FindBeerActivity.java
package com.hfad.beeradviser.Java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.TextView;

import com.hfad.beeradviser.R;

import java.util.List;

public class FindBeerActivity extends AppCompatActivity {
    private BeerExpert expert = new BeerExpert();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_find_beer);

    }

    //Вызывается при щелчке на кнопке
    public void onClickFindBeer(View view) {

        //Получить ссылку на TextView
        TextView brands = (TextView) findViewById(R.id.brands);

        //Получить ссылку на Spinner
        Spinner color = (Spinner) findViewById(R.id.color);

        //Получить вариант, выбранный в Spinner
        String beerType = String.valueOf(color.getSelectedItem());

        //Получить рекомендации от класса BeerExpert
        List<String> brandsList = expert.getBrands(beerType);

        StringBuilder brandsFormatted = new StringBuilder();

        for (String brand : brandsList) {
            brandsFormatted.append(brand).append('\n');
        }

        //Ввыести сорта пива
        brands.setText(brandsFormatted);
    }
}

File: app/src/main/java/com/hfad/beeradviser/Kotlin/FindBeerActivity.kt
package com.hfad.beeradviser.Kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.hfad.beeradviser.R
import kotlinx.android.synthetic.main.activity_find_beer.*
import java.lang.StringBuilder

class FindBeerActivity : AppCompatActivity() {
    private val expert = BeerExpert()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_find_beer)
    }

    //Вызывается при щелчке на кнопке
    fun onClickFindBeer(view: View) {

        //Получить ссылку на TextView
        val brands = brands

        //Получить ссылку на Spinner
        val color = color

        //Получить вариант, выбраннны в Spinner
        val beerType = color.selectedItem.toString()

        //Получить рекомендации от класса BeerExpert
        val brandsList = expert.getBrands(beerType)

        val brandsFormatted = StringBuilder()

        for (brand in brandsList) {
            brandsFormatted.append(brand).append('\n')
        }

        //Вывести сорта пива
        brands.text = brandsFormatted
    }
}
